app/models/user.rb

```diff
@@ -29,6 +29,26 @@ class User < ApplicationRecord
   scope :allowing_liked_event_notification,
         -> { joins(:notification_timings).merge(NotificationTiming.liked_event) }
 
+  has_many :relationships, class_name: "Relationship", foreign_key: "follower_id", dependent: :destroy
+  has_many :reverse_of_relationships, class_name: "Relationship", foreign_key: "followed_id", dependent: :destroy
+
+  has_many :followings, through: :relationships, source: :followed
+  has_many :followers, through: :reverse_of_relationships, source: :follower
+
+  # フォローしたときの処理
+  def follow(user_id)
+    relationships.create(followed_id: user_id)
+  end
+  # フォローを外すときの処理
+  def unfollow(user_id)
+    relationship = relationships.find_by(followed_id: user_id)
+    relationship.destroy if relationship
```

Comment:
```suggestion
   return if relationship.blank?
   relationship.destroy 
```
個人的にはこっち派(直さなくてもいいです)

Reply:
ブシトラさん。

コメントありがとうございます！

参考になります！
ありがとうございます！

Reply:
early return で明示的に抜ける手法ですね
https://zenn.dev/media_engine/articles/early_return